pkg/api/registered/registered.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package to keep track of API Versions that have been registered in api.Scheme.
+package registered
+
+import (
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+// List of registered API versions.
+// The list is in the order of most preferred to the least.
+var RegisteredVersions []string
+
+func init() {
+	// The list of valid API versions to test that KUBE_API_VERSIONS does not contain an invalid version.
+	// Note that the list is in ascending sorted order.
+	validAPIVersions := []string{"v1", "v1beta1", "v1beta2", "v1beta3"}
```

Comment:
nit: a set or map[string]bool is more natural considering what you do with this.